### PR TITLE
New feature: add a new policy for fixing clashes

### PR DESCRIPTION
**Pull request by iblis17**
Hi,
     I add an new command line option for `drive clashes`. Let user switch the policy between *rename* (the original) and *trash*. When I encounter the clashes, I always trash it then re-push again. Hope this also useful for other user. And the simple usage is documented in the commit message.

To be honest, I never wrote any line of golang code before I decide to add this feature. I just browsed the golang tutorial at the last Sunday afternoon. Please read my code carefully and comment. :)

### CODE REVIEW

File: src/clashes.go [hunk 1]
```diff
@@ -20,6 +20,13 @@ import (
 	"strings"
 )
 
+type FixClashesMode uint8
+
+const (
+	FixClashesRename FixClashesMode = iota
```
Comment (odeke-em) [inline src/clashes.go]: Let's make this `1 + iota`, so that when we  can catch the
zero/uninitialized value, and reject it.

File: src/clashes.go [hunk 2]
```diff
@@ -275,3 +287,27 @@ func autoRenameClashes(g *Commands, clashes []*Change) error {
 
 	return composedError
 }
+
+func autoTrashClashes(g *Commands, clashes []*Change) error {
+	for _, c := range clashes {
+		// make c.Op() == OpDelete
```
Comment (odeke-em) [inline src/clashes.go]: Let's just update this comment to
```go
// Let's coerce this change to a deletion
```

File: src/clashes.go [hunk 3]
```diff
@@ -275,3 +287,27 @@ func autoRenameClashes(g *Commands, clashes []*Change) error {
...
+		c.Dest = c.Src
+		c.Src = nil
+	}
+
+	if g.opts.canPrompt() {
+		g.log.Logln("Some clashes found, trashes them ?")
```
Comment (odeke-em) [inline src/clashes.go]: Please replace
`trashes them ?`
with
`trash them?`

**Comment by odeke-em** [comment]
Hello there @iblis17, thank you for the PR and welcome to drive!

This is awesome, your first Go code looks great!
I've made 3 comments for minor feedback otherwise we are in good shape.

File: src/clashes.go (hunk 1 above)
Comment (iblis17) [inline src/clashes.go]: done @ 9d078d9

File: src/clashes.go (hunk 3 above)
Comment (iblis17) [inline src/clashes.go]: done @ 04dcf5e

**Comment by iblis17** [comment]
`README.md` updated @ 283066c

**Comment by odeke-em** [comment]
Dope! One last thing, let's squash all these commits into one and then I'll merge it in. Just in
case you need to
```shell
$ git rebase -i head~6
```

and when prompted for the message, let's set it to something like this
```shell
clashes: add trash and no-prompt to deal with clashes

Added trash and no-prompt to clash. The default is `rename` but
in some cases it is convenient to trash duplicates. This PR makes
that possible.
```

Thank you!

**Comment by iblis17** [comment]
Rebased !

**Comment by odeke-em** [comment]
Thank you, LGTM!

**Comment by jpambrun** [comment]
This change is confusing when using "drive clashes -fix -fix-mode trash [remote folder]" since is trashes both files. When doing so, I expected the newest file to be kept. Deleting both files certainly fixed the clash, but so would "rm -Rf" :wink:.

**Comment by iblis17** [comment]
@jpambrun Trashing them all and re-pushing is just my habit. Feel free to rename it and add the feature you want.

----

Maybe I should notice user that it will trash both of new and old files on `README.md`

**Comment by jpambrun** [comment]
What is the purpose of this? It does not make a lot of sense.

Especially since in my case, I had no local copy.

**Comment by iblis17** [comment]
There are just different use cases. And I think there are no correct one or wrong one.
If this feature is not fitting your requirement. what about adding another function or option, contributing the code back, and make `drive` better!

If you really want that feature and do not want to write code, please open a feature request. I can push it into my TODO stack... Maybe I will make it at some Sunday afternoon.

**Comment by jpambrun** [comment]
I'm sorry if that came out a bit harsh; that was not the intent. I don't feel very strongly about this issue/feature and the proposed changes make it clear that it is a destructive operation. 

Personally, I wouldn't make any destructive operations the default behavior for this type of application. I wanted to fix clashes and I issued "drive clashes -fix -fix-mode trash" on in a clean ~/gdrive folder thinking it would fix clashes based on the documentation.

Anyway, I appreciate your contribution.

**Comment by iblis17** [comment]
> Personally, I wouldn't make any destructive operations the default behavior for this type of application. I wanted to fix clashes and I issued "drive clashes -fix -fix-mode trash" on in a clean ~/gdrive folder thinking it would fix clashes based on the documentation.

Hearing a different need is not a bad thing, and it's the source of innovation of open source. :)

I will try to add a new policy called `trash-old` (or something like that), if I get free time.

**Comment by jpambrun** [comment]
Having different needs is not the issue I was trying to raise. The issue is
with the expectations that my data is not going to be lost by "fixing"
clashes. Deleting them all is not what a typical user would call a "fix".

So basically,yes, having different expectations when it comes to the
potential lost of data is a bad thing.
The better solution would be to call your fix-mode trash-all and the one I
proposed trash.
The is the safest, clearest way to include your feature.

**Comment by iblis17** [comment]
ok, I got it. (~I think I am quite used to *fix* them via C4~

**Pull request merged**